src/main/drivers/pwm_output_dshot_hal.c

```diff
@@ -52,38 +52,92 @@ uint8_t getTimerIndex(TIM_TypeDef *timer)
 void pwmWriteDshotInt(uint8_t index, uint16_t value)
 {
     motorDmaOutput_t *const motor = &dmaMotors[index];
-
+#ifdef USE_DSHOT_DMAR
+    if (!motor->timerHardware || !motor->timerHardware->dmaTimUPRef) {
+        return;
+    }
+#else
     if (!motor->timerHardware || !motor->timerHardware->dmaRef) {
         return;
     }
+#endif
 
     uint16_t packet = prepareDshotPacket(motor, value);
 
     uint8_t bufferSize = loadDmaBuffer(motor, packet);
 
+#ifdef USE_DSHOT_DMAR
+    /* Optimize me */
+    uint8_t channel_index = 0;
+    switch(motor->timerHardware->channel)
+    {
+        case TIM_CHANNEL_1:
+        channel_index = 0;
+        break;
+        case TIM_CHANNEL_2:
+        channel_index = 1;
+        break;
+        case TIM_CHANNEL_3:
+        channel_index = 2;
+        break;
+        case TIM_CHANNEL_4:
+        channel_index = 3;
+        break;
+    }
+    // load channel data into burst buffer
+    for(int i = 0; i< bufferSize; i++) {
+        dmaMotorTimers[motor->timerIndex].dmaBurstBuffer[channel_index]=motor->dmaBuffer[i];
+        channel_index+=4;
+    }
+    if(HAL_DMA_STATE_READY == motor->TimHandle.hdma[motor->timerDmaIndex]->State) {
+        HAL_DMA_Start_IT(motor->TimHandle.hdma[motor->timerDmaIndex], (uint32_t)dmaMotorTimers[motor->timerIndex].dmaBurstBuffer, (uint32_t)&motor->TimHandle.Instance->DMAR, (DSHOT_DMA_BUFFER_SIZE*4));
```

Comment:
`bufferSize * 4` instead of `DSHOT_DMA_BUFFER_SIZE` (protocol may use different number of intervals)

Reply:
Is it OK to trigger DMA here? Only data for single motor are prepared at this point ... (but timer is enabled after all channels)